.github/workflows/build_and_test.yml

```diff
@@ -56,36 +53,51 @@ jobs:
     - name: See if version exists on remote
       id: test_package_exists
       run: |
-        conan remote add tket-libs https://quantinuumsw.jfrog.io/artifactory/api/conan/tket1-libs
+        conan remote add tket-libs https://quantinuumsw.jfrog.io/artifactory/api/conan/tket1-libs --index 0
         tket_package_exists=`conan search -r tket-libs "tket/${{ steps.tket_ver.outputs.tket_ver }}@tket/stable" | grep "not found" > /dev/null 2>&1 && echo false || echo true`
         echo "tket_package_exists=${tket_package_exists}" >> $GITHUB_OUTPUT
     - name: Check tket version bump
       if: github.event_name == 'pull_request' && github.event.pull_request.base.ref == 'develop' && steps.filter.outputs.tket == 'true' && steps.test_package_exists.outputs.tket_package_exists == 'true'
       run: exit 1
 
+  check_docs_tket:
+    name: Check documentation build
+    needs: check_changes
+    if: needs.check_changes.outputs.tket_changed == 'true' || needs.check_changes.outputs.doxyfile_changed == 'true'
+    runs-on: 'ubuntu-22.04'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check doxygen
+        run: |
+          sudo apt install -y doxygen graphviz
+          cd tket && doxygen
+
+  check_format_tket:
+    name: Check C++ code formatting
+    needs: check_changes
+    if: needs.check_changes.outputs.tket_changed == 'true'
+    runs-on: 'macos-12'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check C++ code formatting
+        run: |
+          brew update
+          brew install clang-format@16
+          git ls-files "*.cpp" "*.hpp" | xargs clang-format -style=file --dry-run --Werror
+
   build_test_tket:
     name: Build and test (tket)
     needs: check_changes
+    if: needs.check_changes.outputs.tket_changed == 'true'
     strategy:
       matrix:
-        os: ['ubuntu-22.04', 'macos-12', 'windows-2022']
```

Comment:
Is the windows build so different that it is worth separating, rather than making some steps conditional?

Reply:
Yes, unfortunately, it's completely different. The ccache cache setup has to be done by hand (the ccache action doesn't quite work for windows). Could theoretically do it by hand for the others, but then the commands will be different for bash vs. powershell anyway. Also, the ccache`base_dir` setting doesn't work on windows so have to use conan `install` + `build` for ccache to work.